src/allmydata/test/web/test_web.py

```diff
@@ -1988,7 +1985,8 @@ def test_GET_DIRECTORY_html_filenode_encoding(self):
             self.failUnlessIn(entity, self._htmlname_raw)
             self.failIfIn(entity, self._htmlname_escaped)
         self.failUnlessIn('&', re.sub(r'&(amp|lt|gt|quot|apos);', '', self._htmlname_raw))
-        self.failIfIn('&', re.sub(r'&(amp|lt|gt|quot|apos);', '', self._htmlname_escaped))
+        # escaped string is now `weirdly&#x27;namedfile_iframe /.txt`
+        self.failUnlessIn('&', re.sub(r'&(amp|lt|gt|quot|apos);', '', self._htmlname_escaped))
```

Comment:
On the surface, this change seemed wrong.  Changing from `failIfIn` to `failUnlessIn` exactly reverse the test which must be wrong.  So I dug in more and now I'm not even sure what this is supposed to be testing.

It sure doesn't look like it has anything to do with `test_GET_DIRECTORY_html_filenode_encoding`.  `self.htmlname_escaped` and `self._htmlname_raw` are set by `setUp`.  They are defined as the escaping of some string literal.  What is the point?

It seems to me that lines 1981 through 1989 should just be deleted.  This leaves `self_htmlname_escaped` defined but unused so it can also be deleted.  And `self._htmlname_raw` is defined and used in the definition of some other attributes immediately afterwards - so there seems to be no reason for it to be an attribute of self.

Reply:
I too don't know the point of `self._htmlname_escaped`, `self._htmlname_escaped_attr`, and `self._htmlname_escaped_double`.  I did just enough translation to make the tests pass, without giving it too much thought, assuming that there must be a point.  Perhaps the point was  to ensure that `escapetoXML(self._htmlname_raw)` etc. worked at all?  I mean, `setUp` had been calling `nevow.util.escapeToXML()`, before this PR replaced it with `html.escape()`, but I don't fully understand the reasoning behind doing things this way.

Changing `failIfIn` to `failUnlessIn` did produce some odd code smell.  I just did not pause to think about it. :-/